temporalio/converter.py

```diff
@@ -474,12 +474,21 @@ def encoding(self) -> str:
 
     def to_payload(self, value: Any) -> Optional[temporalio.api.common.v1.Payload]:
         """See base class."""
+        # Check for json callable, and invoke it
+        # This covers Pydantic models.
+        parse_obj_attr = inspect.getattr_static(value, "json", None)
```

Comment:
```suggestion
        json_method = getattr(value, "json", None)
```

And just use that method if present. We need to make sure we work with proxied pydantic models which can happen in a sandbox so we need to not bypass `__getattribute__`.